app/services/instrumentation_support.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+# Methods to support instrumentation
+module InstrumentationSupport
+  # Provides a source annotation for logging queries.
+  # Example: storage_root.moab_records.annotate(source_annotation).sum(:size)
+  # See lib/slow_query_logger.rb
+  def source_annotation
+    return unless Settings.slow_queries.enable
```

Comment:
And I guess if the `annotate` method gets a `nil`, it no-ops and does the right thing?

Reply:
Correct.

Reply:
since this was a question, is it worth a comment or a test?